hudi-common/src/main/java/org/apache/hudi/common/table/view/AbstractTableFileSystemView.java

```diff
@@ -282,6 +286,20 @@ public void reset() {
     }
   }
 
+  /**
+   * Resets the view states, which can be overridden by subclasses.  This reset logic is guarded
+   * by the write lock.
+   * <p>
+   * NOTE: This method SHOULD BE OVERRIDDEN for any custom logic.  DO NOT OVERRIDE
+   * {@link AbstractTableFileSystemView#reset} directly, which may cause stale file system view
+   * to be served.
+   */
+  protected void runReset() {
```

Comment:
Guess the method should be guarded by reset `writeLock` ? `#reset` should invoke `#runReset` instead I guess.

And can we give these methods more straight-forward names?

Like sync -> syncThreadSafely, runSync -> sync, reset->resetThreadSafely, runReset -> reset

Reply:
Yes, `reset()` should call `runReset()`.  Not sure why the change is reverted somehow.

I don't think we should change the naming of `sync` and `reset` in the interface `SyncableFileSystemView` as they can also have a different implementation.  I've added documentation to make sure how the custom logic should be overridden.

Reply:
Don’t think the doc makes any sense, let’s give them better name.

Reply:
Removed `runSync` and `runReset` methods to avoid confusion and make every implementation explicitly use write lock except remote FSV.  If new file system view needs to be added, the author should look at existing implementation for reference.  Renaming won't prevent them the author doing the wrong thing.